lib/resources/postgres_session.rb

```diff
@@ -38,39 +38,37 @@ class PostgresSession < Inspec.resource(1)
       # db: databse == db_user running the sql query
 
       describe sql.query('SELECT * FROM pg_shadow WHERE passwd IS NULL;') do
-        its('output') { should eq('') }
+        its('output') { should eq '' }
       end
     "
-    attr_reader :user, :pass, :host, :db, :escaped_query, :psql_cmd
 
     def initialize(user, pass, host = nil)
       @user = user || 'postgres'
       @pass = pass
       @host = host || 'localhost'
     end
 
-    def escaped_query(query)
-      return warn 'no query specificed' if query.empty? || query.nil?
-      Shellwords.escape(query)
-    end
-
-    def create_psql_cmd(query, db = [])
-      @db = db
-      dbs = db.map { |x| "-d #{x}" }.join(' ')
-      @escaped_query = escaped_query(query)
-      "PGPASSWORD='#{@pass}' psql -U #{@user} #{dbs} -h #{@host} -A -t -c #{@escaped_query}"
-    end
-
     def query(query, db = [])
-      @db = db
-      @psql_cmd = create_psql_cmd(query, @db)
-      cmd = inspec.command(@psql_cmd)
+      psql_cmd = create_psql_cmd(query, db)
+      cmd = inspec.command(psql_cmd)
       out = cmd.stdout + "\n" + cmd.stderr
       if cmd.exit_status != 0 || out =~ /could not connect to .*/ || out.downcase =~ /^error:.*/
         skip_resource "Can't read run query #{query.inspect} on postgres_session: #{out}"
       else
         Lines.new(cmd.stdout.strip, "PostgreSQL query: #{query}")
       end
     end
+
+    private
+
+    def escaped_query(query)
+      Shellwords.escape(query)
+    end
+
+    def create_psql_cmd(query, db = [])
+      dbs = db.map { |x| "-d #{x}" }.join(' ')
+      @escaped_query = escaped_query(query)
```

Comment:
This line can be eliminated, correct? You're not using this variable anywhere, and you're calling `escaped_query` directly in the returned string in the next line.